validation/controllers_test.go

```diff
@@ -345,3 +345,211 @@ func FuzzAdvanced(f *testing.F) {
+		{"K e1 b", brdBlackUsFEN, "e1", 403, ""},
+
+		{"R h1 w", brdWhiteUsFEN, "7", 200, `{"moves":[5,6,15]}`},
+		{"R h1 b", brdBlackUsFEN, "7", 403, ""},
+
+		{"p b2 w", brdWhiteUsFEN, "b2", 403, ""},
+		{"p b2 b", brdBlackUsFEN, "b2", 200, `{"moves":[0]}`},
+
+		{"P d2 w", brdWhiteUsFEN, "11", 403, ""},
+		{"P d2 b", brdBlackUsFEN, "11", 403, ""},
+
+		{"N f2 w", brdWhiteUsFEN, "f2", 403, ""},
+		{"N f2 b", brdBlackUsFEN, "f2", 403, ""},
+
+		{"n b3 w", brdWhiteUsFEN, "17", 403, ""},
+		{"n b3 b", brdBlackUsFEN, "17", 200, `{"moves":[0,2,11,27,32]}`},
+
+		{"Q d3 w", brdWhiteUsFEN, "d3", 200,
+			`{"moves":[5,10,12,17,18,20,21,22,26,27,28,33,35,40,43]}`},
+		{"Q d3 b", brdBlackUsFEN, "d3", 403, ""},
+
+		{"P h3 w", brdWhiteUsFEN, "23", 403, ""},
+		{"P h3 b", brdBlackUsFEN, "23", 403, ""},
+
+		{"b b4 w", brdWhiteUsFEN, "b4", 403, ""},
+		{"b b4 b", brdBlackUsFEN, "b4", 200, `{"moves":[11,16,18,32]}`},
+
+		{"P g4 w", brdWhiteUsFEN, "30", 403, ""},
+		{"P g4 b", brdBlackUsFEN, "30", 403, ""},
+
+		{"q h4 w", brdWhiteUsFEN, "h4", 403, ""},
+		{"q h4 b", brdBlackUsFEN, "h4", 200,
+			`{"moves":[13,22,23,30,39,47,55,63]}`},
+
+		{"k c5 w", brdWhiteUsFEN, "34", 403, ""},
+		{"k c5 b", brdBlackUsFEN, "34", 200, `{"moves":[41]}`},
+
+		{"P f5 w", brdWhiteUsFEN, "f5", 200, `{"moves":[45,46]}`},
+		{"P f5 b", brdBlackUsFEN, "f5", 403, ""},
+
+		{"p g5 w", brdWhiteUsFEN, "38", 403, ""},
+		{"p g5 b", brdBlackUsFEN, "38", 403, ""},
+
+		{"n c6 w", brdWhiteUsFEN, "c6", 403, ""},
+		{"n c6 b", brdBlackUsFEN, "c6", 200,
+			`{"moves":[27,32,36,52,57]}`},
+
+		{"r d6 w", brdWhiteUsFEN, "43", 403, ""},
+		{"r d6 b", brdBlackUsFEN, "43", 200,
+			`{"moves":[19,27,35,44,45,46,47,51]}`},
+
+		{"p a7 w", brdWhiteUsFEN, "a7", 403, ""},
+		{"p a7 b", brdBlackUsFEN, "a7", 200, `{"moves":[32,40]}`},
+
+		{"P e7 w", brdWhiteUsFEN, "52", 200, `{"moves":[59,60]}`},
+		{"P e7 b", brdBlackUsFEN, "52", 403, ""},
+
+		{"B f7 w", brdWhiteUsFEN, "f7", 200,
+			`{"moves":[17,26,35,39,44,46,60,62]}`},
+		{"B f7 b", brdBlackUsFEN, "f7", 403, ""},
+
+		{"r d8 w", brdWhiteUsFEN, "59", 403, ""},
+		{"r d8 b", brdBlackUsFEN, "59", 200,
+			`{"moves":[51,56,57,58,60,61,62,63]}`},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			u, _ := url.Parse(srv.URL)
+			v := u.Query()
+			v.Add("board", tc.brd)
+			v.Add("from", tc.from)
+			u.RawQuery = v.Encode()
+
+			for _, method := range []string{"GET", "HEAD", "POST"} {
+				client := &http.Client{}
+				req, _ := http.NewRequest(method, u.String(), nil)
+				res, err := client.Do(req)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				switch method {
+				case "GET", "HEAD":
+					if tc.resultStatus != res.StatusCode {
+						t.Fatalf("want %v, got %s", tc.resultStatus, res.Status)
+					}
+				case "POST":
+					if res.StatusCode != http.StatusMethodNotAllowed {
+						t.Fatalf("unexpected reply for method %s: %s", method, res.Status)
+					}
+				}
```

Comment:
Тут нет проверки на `400`, фаззер находит :)

Reply:
Кстати, в Simple он тоже это находит.

Reply:
Знаешь, я бы, наверно, попробовал объединить три фаззера в один (запустить три тест-сервера, и итерироваться по ним в `range`, как это сделано для методов, к примеру) — там ведь логика одинаковая, считай. Тогда не будет таких упущений…

Reply:
объединила три фаззера в один

Reply:
Честно говоря, я не понимаю, почему сервер получает 400 ответ при методе Post, если у меня в хендлере четко прописано, что при любом методе, который не Get и не Head должно приходить 405.

Это особенность client.Do?
Они пишут Generally Get, Post, or PostForm will be used instead of Do.